code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -536,12 +536,13 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 		if(TOUCH)
 			for(var/datum/reagent/R in reagent_list)
 				if(ismob(A))
-					var/datum/reagent/MR = A.reagents.get_reagent_by_type(R.type)
-					if(!mob_reagent_threshold || !MR || (mob_reagent_threshold > MR.volume))
-						if(isanimal(A))
-							R.reaction_animal(A, TOUCH, R.volume+volume_modifier)
-						else
-							R.reaction_mob(A, TOUCH, R.volume+volume_modifier, remove_reagents)
+					if(A.reagents)
```

Comment:
are you sure this indentation is right

Reply:
Yes, it checks ismob and then checks if A has reagents, because the rest of the proc involves a check that includes the existence of reagents

Reply:
anyway I think I made it better now